src/TodosContext.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useMemo, useState } from 'react';
+import { Todo } from './types/Todo';
+import { TodosContextType } from './types/TodosContext';
+import { Status } from './types/Status';
+import { ErrorMessage } from './types/ErrorMessage';
+
+export const TodosContext = React.createContext<TodosContextType>({
+  todos: [],
+  setTodos: () => {},
+  visibleTodos: [],
+  filter: Status.ALL,
+  setFilter: () => { },
+  isChecked: false,
+  setIsChecked: () => {},
+  errorMessage: ErrorMessage.DEFAULT,
+  setErrorMessage: () => {},
+  isProcessing: [],
+  setIsProcessing: () => {},
+});
+
+type Props = {
+  children: React.ReactNode;
+};
+
+export const TodoProvider: React.FC<Props> = ({ children }) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isChecked, setIsChecked] = useState<boolean>(false);
```

Comment:
there is no reason to keep data in state if it can be calculated based on existing ones

Reply:
I did it this way to keep the data in a state so that its changes were synchronized between all components.